pkg/networkserver/grpc_gsns.go

```diff
@@ -385,14 +376,17 @@ func (ns *NetworkServer) matchAndHandleDataUplink(ctx context.Context, dev *ttnp
 		dev.MACState = macState
 		dev.Session.StartedAt = up.ReceivedAt
 
+		fCntGap = cmacFMatchResult.FullFCnt
```

Comment:
Doesn't sound like a gap anymore?

Reply:
It does, the gap from 0 to value of uplink frame counter.
We need to compare the gap from previous value to new one according to the spec - for FCnt resets the previous value is 0.